include/unifex/type_erased_stream.hpp

```diff
@@ -104,24 +104,36 @@ struct _stream<Values...>::type final {
     virtual void start_cleanup(cleanup_receiver_base& receiver) noexcept = 0;
   };
 
+  struct next_op_base {
+    virtual ~next_op_base() = default;
+    virtual bool complete() noexcept = 0;
+  };
+
   template <typename Receiver>
   struct _next_receiver final {
     struct type final : next_receiver_base {
       UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+      next_op_base* op_;
 
-      explicit type(Receiver&& receiver)
-        : receiver_((Receiver &&) receiver) {}
+      explicit type(Receiver&& receiver, next_op_base* op)
+        : receiver_((Receiver &&) receiver), op_(op) {}
```

Comment:
It occurs to me that you could save on the virtual function overhead if you passed the address of the `atomic<char> refcount` instead of the whole op state.